### DISCUSSION TITLE
General Why?-Question

### DISCUSSION THREAD

**Discussion by dekiesel** [technyon/nuki_hub#522] [General]
Hi,

I am looking into how to best set up the Nuki that I plan on ordering. From my understanding Nuki officially supports MQTT. What extra goodies does nuki hub add here? I plan on using Thread and from my understanding nuki hub still lets the lock connect to the cloud so the lock can update its firmware?

This is definitely a beginners question and I apologize in advance if it is stupid, but I welcome every help here :)

Thanks!

**Comment by iranl** [comment]
The official Nuki MQTT implementation is fast in communicating the lockstate and accepting locking actions **if and when it works**
Many reliability issues with the official MQTT implementation have been reported on the Nuki developer forums though.
If the official MQTT implementation works reliably for you (it does for me using a 4.0 and 4.0 pro lock over Thread) you should definitely use it.

This does not mean NukiHub does not have an added use for most, as beside the reliability issues the offical MQTT implementation is also very limited. 
To start it does not support the opener and SSL MQTT.
It also lacks many, many features that are only viewable and changable using the Nuki app and other Nuki API's such as authorizations, logs, keypad, timecontrol, changing device settings etc.

With NukiHub you can automate changing almost anything that you can change using the Nuki App and allows your smart home system to react on much more comprehensive information about your lock, keypad, status changes etc.

NukiHub also supports a hybrid mode where it proxies lockstate updates and actions over the official MQTT implementation for added speed, but uses Bluetooth for all the features that are not supported by the official implementation.

**Reply by dekiesel** [reply]
Thank you **very** much!
